core/controllers/base_test.py

```diff
@@ -1423,3 +1469,393 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
+                set(handler_request_methods) - set(methods_defined_in_schema))
+            if missing_schema_keys:
+                handlers_with_missing_request_schema_keys.append(
+                    handler_class_name)
+                self.log_line(
+                    'Missing keys in HANDLER_ARGS_SCHEMAS for %s: %s.' % (
+                        handler_class_name, ', '.join(missing_schema_keys)))
+
+        error_msg = (
+            'Missing schema keys in HANDLER_ARGS_SCHEMAS for [ %s ] classes.'
+            '\nVisit %s to learn how to write schemas for handler args.' % (
+                ', '.join(handlers_with_missing_request_schema_keys),
+                    self.wiki_page_link))
+
+        self.assertEqual(
+            handlers_with_missing_request_schema_keys, [], error_msg)
+
+    def test_default_value_in_schema_conforms_with_schema(self):
+        """This test checks whether the default_value provided in schema
+        conforms with the rest of the schema.
+        """
+        handlers_with_non_conforming_default_schemas = []
+        list_of_routes_which_need_schemas = (
+            self._get_list_of_routes_which_need_schemas())
+
+        for route in list_of_routes_which_need_schemas:
+            handler = route.handler
+
+            handler_class_name = handler.__name__
+            if handler_class_name in self.handler_class_names_with_no_schema:
+                continue
+            if handler.HANDLER_ARGS_SCHEMAS is None:
+                continue
+
+            schemas = handler.HANDLER_ARGS_SCHEMAS
+            for request_method, request_method_schema in schemas.items():
+                for arg, schema in request_method_schema.items():
+                    if 'default_value' not in schema:
+                        continue
+                    default_value = {arg: schema['default_value']}
+                    default_value_schema = {arg: schema}
+
+                    errors = payload_validator.validate(
+                        default_value, default_value_schema, True)
+                    if len(errors) == 0:
+                        continue
+                    self.log_line(
+                        'Handler: %s, argument: %s, default_value '
+                            'validation failed.' % (handler_class_name, arg))
+
+                    if (handler_class_name not in
+                            handlers_with_non_conforming_default_schemas):
+                        handlers_with_non_conforming_default_schemas.append(
+                            handler_class_name)
+
+        error_msg = (
+            'Schema validation for default values failed for handlers: [ %s ].'
+            '\nVisit %s to learn how to write schemas for handler args.' % (
+                ', '.join(handlers_with_non_conforming_default_schemas),
+                        self.wiki_page_link))
+
+        self.assertEqual(
+            handlers_with_non_conforming_default_schemas, [], error_msg)
+
```

Comment:
@seanlip May be there is a little confusion, I want to take this opportunity to explain some points about the constants used in payload_validator file.
**SCHEMA_REQUIRING_HANDLER_CLASS_NAMES**: Those handlers which needs schema but currently they do not have schemas defined because of the task to add schemas incrementally.
**HANDLER_CLASS_NAMES_WHICH_DO_NOT_REQUIRE_SCHEMAS**: Those handlers which may or maynot be the child classes of BaseHandler and they do not require any schema (eg. WarmupPage or SessionBeginHandler)
**HANDLER_CLASS_NAMES_WITH_NO_SCHEMA** = ( SCHEMA_REQUIRING_HANDLER_CLASS_NAMES + HANDLER_CLASS_NAMES_WHICH_DO_NOT_REQUIRE_SCHEMAS)
In many of the places it is referenced bcause they require the names of all the handlers which do not have schemas defined.
Thus this test is checking if any handler have had schemas then it should not be present in SCHEMA_REQUIRING_HANDLER_CLASS_NAMES list.
 Looking forward for your suggestions.

Reply:
> No, this is still stated wrongly. There is no such list as you mentioned in the docstring.
I've already suggested a rewording of the last part of the docstring in my previous comment. Try to rename the test so that it actually matches what you're testing; work with your mentor if you're not sure how to do this.

In context of this comment.

Reply:
I think the problem I'm having is this: the test does not seem to be referring at all to the "SCHEMA_REQUIRING_HANDLER_CLASS_NAMES list in payload validator". On further investigation this seems to be hidden in self.handler_class_names_with_no_schema, which is not so great because the underlying implementation could change.

Could you use SCHEMA_REQUIRING_HANDLER_CLASS_NAMES directly within the test?

Reply:
Updated, ptal.